code-challenges.js

```diff
@@ -48,28 +124,84 @@ var vowelTester2 = "throw"
+        return "Can't divide by 0! Your console will explode"
+    }
+    //creating four else if statements that see if operator is +, -, /, *
+    else if (typeof num1 === "number" && typeof num2 === "number" && mathOperator === "+") {
+        return num1 + num2
+    }
+    else if (typeof num1 === "number" && typeof num2 === "number" && mathOperator === "-") {
+        return num1 - num2
+    }
+    else if (typeof num1 === "number" && typeof num2 === "number" && mathOperator === "*") {
+        return num1 * num2
+    }
+    else if (typeof num1 === "number" && typeof num2 === "number" && mathOperator === "/") {
+        return num1 / num2
+    }
+    // creating a catch all in case arguments dont match paramenters
+    else {
+        return "Please make sure you enter a number, an operator, and another number"
+    }
+}
 // Uncomment and use the following console logs to test your function
-// console.log(calculator(3, "*", 9))
+console.log(calculator(3, "*", 9))
 // Expected output: 27
 
-// console.log(calculator(16, "+", 3))
+console.log(calculator(16, "+", 3))
 // Expected output: 19
 
-// console.log(calculator(89, "/", 0))
+console.log(calculator(89, "/", 0))
 // Expected output: "Can't divide by 0!"
 
+console.log(calculator("hi", "+", 0))
+
 
 
 // --------------------6) Create a function that takes in the following variable and returns only the websites that end in ".io".
 
-var websites = ["codepen.io", "codecademy.com", "coursera.org", "codepen.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
+//create a function that takes in an array of strings
+
+const ioChecker = (array) => {
+    // filter each value in the array
+    let ioOnly = array.filter(url => {
+        // for each url, lowercase value
+        // for the filter method, only return urls if they include ".io"
+        return url.toLowerCase().includes(".io")
+    }).join(" ").toLowerCase()
+    // joining array into strings
+    // the string
+    return (ioOnly)
+}
+
+var websites = ["CODEPEN.IO", "CODECADEMY.COM", "coursera.org", "paper.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
 // Expected output: "codepen.io" "codepen.io"
+console.log(ioChecker(websites));
+
 
 
 
+// --------------------7) STRETCH: Write a function that takes a number as an argument and uses a WHILE loop to count up to that number from 0
 
-// --------------------6) STRETCH: Write a function that takes a number as an argument and uses a WHILE loop to count up to that number from 0.
+//write a function that takes a number
+const counter = (number) => {
+    // start counter at 0
+    let i = 0
+    // while i is less than number +1
+    while (i < number + 1) {
+        //log i
+        console.log(i);
+        //increment by i
+        i++
+    }
+//uses while loop to coubt up to that number from o
+}
+// Sorry, idk if this is cheating LOL I wasn't able to print with a console.log on 207, so i included the console.log on 200 instead
+counter(10);
```

Comment:
So unfortunately this is not going to function in a production environment correctly, as its not returning anything. The console.log would only ever appear in the terminal, or the console of the web browser. 

But! Take a look at my re-factor below and let me know if you can spot where the changes are to make this work:

`const counter = (number) => {`
    `// start counter at 0`
 `   let i = 0`
   ` // while i is less than number +1`
    `while (i < number) {`
       ` //increment by i`
    `    i++`
  `  }`
  `  return i`
/`/uses while loop to coubt up to that number from o`
`}`

`console.log(counter(10));`